### PR TITLE
Adding support for ceph logs

### PR DESCRIPTION
**Pull request by tintumathew10**
# Description

Adding support for capturing ceph logs.

### CODE REVIEW

File: cephci/cluster_info.py [hunk 1]
```diff
@@ -15,13 +17,16 @@
...
+
+log = Log(__name__)
 
 doc = """
 Utility to gather cluster information
 
     Usage:
        cephci/cluster_info.py (--reuse <FILE>)
-           [--output <YAML>]
+           (--output <YAML>) (--ceph-logs <true/false>)
```
Comment (vamahaja) [inline cephci/cluster_info.py:29]: Use next line for new parameter

File: cephci/cluster_info.py [hunk 2]
```diff
@@ -152,12 +157,44 @@ def gather_info(cluster):
     return info
 
 
+def get_ceph_logs(cluster, log_dir):
+    """Capture ceph cluster logs"""
+    for node in cluster.get_nodes():
+        source_dir = "/var/log/ceph"
```
Comment (vamahaja) [inline cephci/cluster_info.py:163]: this should be global variable with name `CEPH_LOG_DIR`

File: cephci/cluster_info.py [hunk 3]
```diff
@@ -152,12 +157,44 @@ def gather_info(cluster):
...
+        download_dir = f"{log_dir}/ceph_logs/{node.hostname}"
```
Comment (vamahaja) [inline cephci/cluster_info.py:164]: Use `os.path.join` for creating paths

File: cephci/cluster_info.py [hunk 4]
```diff
@@ -152,12 +157,44 @@ def gather_info(cluster):
...
+        os.makedirs(download_dir, exist_ok=True)
+        files = node.get_dir_list(dir_path=source_dir, sudo=True)
+        attributes = node.get_listdir_attr(dir_path=source_dir, sudo=True)
+        log.info(f"files : {files} \nattributes : {attributes}")
```
Comment (vamahaja) [inline cephci/cluster_info.py:168]: Use meaningfule logging message

File: cephci/cluster_info.py [hunk 5]
```diff
@@ -167,12 +204,20 @@ def write_output(data, output):
     args = docopt(doc)
 
     # Get user parameters
-    config, output = args.get("--reuse"), args.get("--output")
+    config, output, ceph_logs = (
```
Comment (vamahaja) [inline cephci/cluster_info.py:207]: Instead use separate lines for each parameter

File: cephci/cluster_info.py [hunk 6]
```diff
@@ -167,12 +204,20 @@ def write_output(data, output):
...
+    )
 
     # Collect cluster information
     _dict, _info = _load_cluster_config(config), {}
     for name in _dict:
-        _info[name] = gather_info(_dict.get(name))
+        _info = gather_info(_dict.get(name))
+
+        # Write data to stream
+        _output = f"{output}/cluster_info_{name}.yaml"
```
Comment (vamahaja) [inline cephci/cluster_info.py:219]: same as above

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: cephci/cluster_info.py (hunk 1 above)
Comment (tintumathew10) [reply cephci/cluster_info.py]: done

File: cephci/cluster_info.py (hunk 4 above)
Comment (tintumathew10) [reply cephci/cluster_info.py]: done

File: cephci/cluster_info.py [hunk 7]
```diff
@@ -152,12 +160,43 @@ def gather_info(cluster):
     return info
 
 
+def get_ceph_logs(cluster, log_dir):
```
Comment (vamahaja) [inline cephci/cluster_info.py:163]: Change name to `get_ceph_var_logs`

File: cephci/cluster_info.py [hunk 8]
```diff
@@ -152,12 +160,43 @@ def gather_info(cluster):
...
+    """Capture ceph cluster logs"""
+    for node in cluster.get_nodes():
+        download_dir = os.path.join(log_dir, "ceph_logs", node.hostname)
+        os.makedirs(download_dir, exist_ok=True)
+        files = node.get_dir_list(dir_path=CEPH_VAR_LOG_DIR, sudo=True)
+        attributes = node.get_listdir_attr(dir_path=CEPH_VAR_LOG_DIR, sudo=True)
+        log.info(f"List ceph var log files : {files}")
+        for file, attribute in zip(files, attributes):
+            if attribute.filename == file:
```
Comment (vamahaja) [inline cephci/cluster_info.py:172]: Why this check required ?

File: cephci/cluster_info.py [hunk 9]
```diff
@@ -152,12 +160,43 @@ def gather_info(cluster):
...
+                if S_ISDIR(attribute.st_mode):
+                    os.makedirs(os.path.join(download_dir, file), exist_ok=True)
+                    ceph_files = node.get_dir_list(
+                        dir_path=os.path.join(CEPH_VAR_LOG_DIR, file), sudo=True
+                    )
+                    for ceph_file in ceph_files:
+                        node.download_file(
```
Comment (vamahaja) [inline cephci/cluster_info.py:179]: Instead of calling download method twice, I would sugest to collect filenames first and download all files

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: cephci/cluster_info.py (hunk 9 above)
Comment (tintumathew10) [reply cephci/cluster_info.py:179]: files are getting downloaded 1 by 1

File: cephci/cluster_info.py (hunk 8 above)
Comment (tintumathew10) [reply cephci/cluster_info.py:172]: there was no way to get filepath and file attribute together in a response, both function has only filename in common, so used this way to compare.

File: cephci/cluster_info.py [hunk 10]
```diff
@@ -152,12 +160,42 @@ def gather_info(cluster):
     return info
 
 
+def get_ceph_var_logs(cluster, log_dir):
+    """Capture ceph cluster var logs"""
+    for node in cluster.get_nodes():
+        download_dir = os.path.join(log_dir, "ceph_logs", node.hostname)
+        os.makedirs(download_dir, exist_ok=True)
+        files = node.get_dir_list(dir_path=CEPH_VAR_LOG_DIR, sudo=True)
```
Comment (vamahaja) [inline cephci/cluster_info.py:168]: Unused variable

File: cephci/cluster_info.py [hunk 11]
```diff
@@ -152,12 +160,42 @@ def gather_info(cluster):
     return info
 
 
+def get_ceph_var_logs(cluster, log_dir):
+    """Capture ceph cluster var logs"""
```
Comment (vamahaja) [inline cephci/cluster_info.py:164]: Add appropriate comments

**Review by vamahaja — CHANGES_REQUESTED** [review]

**Review by vamahaja — APPROVED** [review]

**Pull request merged**